src/views/forms/FormTransactionBase/FormTransactionBase.ts

```diff
@@ -177,8 +182,13 @@ export class FormTransactionBase extends Vue {
     /**
      * it creates the deadlines for the transactions.
      */
-    protected createDeadline(): Deadline {
-        return Deadline.create(this.epochAdjustment);
+    protected createDeadline(deadlineInHours = 2): Deadline {
+        this.$store.dispatch('network/SET_TRANSACTION_DEADLINE', deadlineInHours);
```

Comment:
I believe you need `await` for this because it return promises?

Reply:
It gets triggered on transaction form change for each field, so I don't think it needs to have the await function, so you'll already have the deadline.